DuckDuckGo/AppDelegate.swift

```diff
@@ -386,13 +388,37 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         beginAuthentication()
         autoClear?.applicationWillMoveToForeground()
         showKeyboardIfSettingOn = true
+        syncService.scheduler.resumeSyncQueue()
     }
 
     func applicationDidEnterBackground(_ application: UIApplication) {
         displayBlankSnapshotWindow()
         autoClear?.applicationDidEnterBackground()
         lastBackgroundDate = Date()
         AppDependencyProvider.shared.autofillLoginSession.endSession()
+        suspendSync()
+    }
+
+    private func suspendSync() {
+        if syncService.isSyncInProgress {
+            os_log(.debug, log: .syncLog, "Sync is in progress. Starting background task to allow it to gracefully complete.")
+
+            var taskID: UIBackgroundTaskIdentifier!
+            taskID = UIApplication.shared.beginBackgroundTask(withName: "Cancelled Sync Completion Task") {
+                os_log(.debug, log: .syncLog, "Forcing background task completion")
+                UIApplication.shared.endBackgroundTask(taskID)
```

Comment:
There's really nothing we can do here, besides just preventing the OS from killing the app. After this timeout happens, the resumed app has sync in progress, but the sync operation ended for me right after the app was brought to foreground.